member_map_internal_test.go

```diff
@@ -0,0 +1,256 @@
+
+	// if both have same incarnation, latter member(existing member) will be returned
+	m3 := Member{
+		ID: MemberID{
+			ID: "1",
+		},
+		Status:      Alive,
+		Incarnation: 1,
+	}
+
+	m4 := Member{
+		ID: MemberID{
+			ID: "2",
+		},
+		Status:      Suspected,
+		Incarnation: 1,
+	}
+
+	assert.Equal(t, override(m3, m4), m4)
+}
+
+// 3. {Suspect Ml, inc=i} overrides {Suspect Ml, inc=j}, i>j
+func TestMember_override_suspect_alive(t *testing.T) {
+
+	m1 := Member{
+		Status:      Suspected,
+		Incarnation: 2,
+	}
+
+	m2 := Member{
+		Status:      Alive,
+		Incarnation: 1,
+	}
+
+	assert.Equal(t, override(m1, m2), m1)
+	assert.Equal(t, override(m2, m1), m1)
+
+	// if both have same incarnation, latter member(existing member) will be returned
+	m3 := Member{
+		ID: MemberID{
+			ID: "1",
+		},
+		Status:      Alive,
+		Incarnation: 1,
+	}
+
+	m4 := Member{
+		ID: MemberID{
+			ID: "2",
+		},
+		Status:      Suspected,
+		Incarnation: 1,
+	}
+
+	assert.Equal(t, override(m3, m4), m4)
+}
+
+// 4. {Suspect Ml, inc=i} overrides {Alive Ml, inc=j}, i>=j
+func TestMember_override_suspect_suspect(t *testing.T) {
+
+	m1 := Member{
+		Status:      Suspected,
+		Incarnation: 2,
+	}
+
+	m2 := Member{
+		Status:      Alive,
+		Incarnation: 1,
+	}
+
+	assert.Equal(t, override(m1, m2), m1)
+	assert.Equal(t, override(m2, m1), m1)
+
+	// if both have same incarnation, Suspect will be returned
+	m3 := Member{
+		ID: MemberID{
+			ID: "1",
+		},
+		Status:      Suspected,
+		Incarnation: 1,
+	}
+
+	m4 := Member{
+		ID: MemberID{
+			ID: "2",
+		},
+		Status:      Alive,
+		Incarnation: 1,
+	}
+
+	assert.Equal(t, override(m3, m4), m3)
+}
+
+// 5.  {Dead Ml, inc=i} overrides {Suspect Ml, inc=j}, any j
+func TestMember_override_dead_alive(t *testing.T) {
+
+	m1 := Member{
+		Status:      Dead,
+		Incarnation: 2,
+	}
+
+	m2 := Member{
+		Status:      Suspected,
+		Incarnation: 1,
+	}
+
+	assert.Equal(t, override(m1, m2), m1)
+	assert.Equal(t, override(m2, m1), m1)
+
+	// if both have same incarnation, latter member(existing member) will be returned
+	m3 := Member{
+		ID: MemberID{
+			ID: "1",
+		},
+		Status:      Dead,
+		Incarnation: 1,
+	}
+
+	m4 := Member{
+		ID: MemberID{
+			ID: "2",
+		},
+		Status:      Suspected,
+		Incarnation: 1,
+	}
+
+	assert.Equal(t, override(m3, m4), m4)
+}
+
+// 6. {Dead Ml, inc=i} overrides {Alive Ml, inc=j}, any j
+func TestMember_override_dead_suspect(t *testing.T) {
+
+	m1 := Member{
+		Status:      Dead,
+		Incarnation: 2,
+	}
+
+	m2 := Member{
+		Status:      Alive,
+		Incarnation: 1,
+	}
+
+	assert.Equal(t, override(m1, m2), m1)
+	assert.Equal(t, override(m2, m1), m1)
+
+	// if both have same incarnation, latter member(existing member) will be returned
+	m3 := Member{
+		ID: MemberID{
+			ID: "1",
+		},
+		Status:      Dead,
+		Incarnation: 1,
+	}
+
+	m4 := Member{
+		ID: MemberID{
+			ID: "2",
+		},
+		Status:      Alive,
+		Incarnation: 1,
+	}
+
+	assert.Equal(t, override(m3, m4), m4)
+}
+
```

Comment:
I think there's no test cases when 

* {Confirm Ml, inc=i} overrides
    - {Alive Ml, inc=j}, any j
    - {Alive Ml, inc=j}, any j

for **any j** cases